tests/Execution/string-literals.java

```diff
@@ -21,5 +21,8 @@ public static void main(String[] args)
         // CHECK: 1
         var hello2 = "Hello World";
         print(hello1 == hello2);
+
+        // CHECK: 1
+        print(hello1.intern() == hello2.intern());
```

Comment:
Since `hello1` and `hello2` are both from the same string literal (value-wise), they are probably already interned and the `intern` call itself does not do anything in their case.

I think it'd be nice to add a test where we have two strings with equal value that aren't interned (I'd `print` and check that), and then are equal after interning. You should be able to construct such a string via concattenation with `StringBuilder` e.g.